apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -143,58 +168,186 @@ where
             .get_protocol_key()
             .expect("Validators should always have a protocol key");
 
-        iter_protocol_txs(VoteExtensionDigest {
+        let txs: Vec<_> = iter_protocol_txs(VoteExtensionDigest {
             ethereum_events,
             validator_set_update,
         })
         .map(|tx| tx.sign(protocol_key).to_bytes())
-        .collect()
+        .collect();
+
+        // TODO(feature = "abcipp"):
+        // - alloc space for each protocol tx
+        // - handle space allocation errors
+        // - transition to new allocator state
+
+        todo!()
     }
 
     /// Builds a batch of vote extension transactions, comprised of Ethereum
     /// events and, optionally, a validator set update
     #[cfg(not(feature = "abcipp"))]
-    fn build_vote_extension_txs(&mut self, txs: &[TxBytes]) -> Vec<TxBytes> {
-        if self.storage.last_height != BlockHeight(0) {
-            deserialize_vote_extensions(txs).collect()
-        } else {
+    fn build_vote_extension_txs(
```

Comment:
I think it would be more consistent for this to be named `build_protocol_txs` now

Reply:
sure

Reply:
note to self: probably also rename `build_mempool_txs` to `build_encrypted_txs`, since we also fetch protocol txs from the mempool